cmake/tpl/FindPETSC.cmake

```diff
@@ -13,764 +11,109 @@
-    endif()
-  endforeach()
-
 endif()
+pkg_check_modules(PKG_PETSC "PETSc${_pkg_version_spec}")
+unset(_pkg_version_spec)
+
+# Find the PETSC libraries
+set(_petsc_libs )
+foreach(_next_lib IN LISTS PKG_PETSC_LIBRARIES)
+  find_library(_petsc_lib_${_next_lib} NAMES ${_next_lib} HINTS ${PKG_PETSC_LIBRARY_DIRS})
+  if(_petsc_lib_${_next_lib})
+    list(APPEND _petsc_libs "${_petsc_lib_${_next_lib}}")
+  endif()
+endforeach()
 
-# Set which state variables to check to determine if the PETSC configuration is
-# current and clear the other state variables
-if (PETSC_INCLUDES OR PETSC_LIBRARIES)
-
-  if (PETSC_INCLUDES AND PETSC_LIBRARIES)
-
-    set (PETSC_STATES "LIBRARIES;INCLUDES" CACHE INTERNAL "" FORCE)
-    set (PETSC_DIR  "" CACHE PATH "Path to the root of a PETSc installation" FORCE)
-    set (PETSC_ARCH "" CACHE STRING "PETSc architecture" FORCE)
-
-  else ()
-
-    string (CONCAT msg
-      "Both PETSC_INCLUDES and PETSC_LIBRARIES must be provided:\n"
-      "  PETSC_INCLUDES=${PETSC_INCLUDES}\n"
-      "  PETSC_LIBRARIES=${PETSC_LIBRARIES}")
-    message (FATAL_ERROR ${msg})
-
-  endif ()
-
-else ()
-
-  set (PETSC_STATES "DIR;ARCH" CACHE INTERNAL "" FORCE)
-  set (PETSC_INCLUDES  "" CACHE STRING "Semi-colon separated list of PETSc include directories" FORCE)
-  set (PETSC_LIBRARIES "" CACHE STRING "Semi-colon separated list of PETSc link libraries" FORCE)
-
-endif ()
-
-# Keep track of FindPETSC state so that we do not do the complete
-# set of tests and variable lookups every time cmake is run.
-include (FindPackageMultipass)
-set (petsc_slaves LIBRARIES_SYS LIBRARIES_VEC LIBRARIES_MAT LIBRARIES_DM LIBRARIES_KSP LIBRARIES_SNES LIBRARIES_TS)
-set (petsc_deps LIBRARY_DIR INCLUDE_DIR LIBRARIES_ INCLUDES_ COMPILER MPIEXEC EXECUTABLE_RUNS ${petsc_slaves})
-find_package_multipass (PETSC petsc_config_current STATES ${PETSC_STATES} DEPENDENTS ${petsc_deps})
-
-# This runs anytime the current configuration is not current.
-# This happens either when a user sets PETSC_CURRENT=FALSE,
-# or when one of the dependents given to find_package_multipass changes.
-if (NOT petsc_config_current)
-
-  if (PETSC_INCLUDES AND PETSC_LIBRARIES)
-
-    message (STATUS "Finding PETSC using PETSC_INCLUDES and PETSC_LIBRARIES")
-
-    # extract path from PETSC_INCLUDES
-    foreach (_include_dir ${PETSC_INCLUDES})
-      if (EXISTS "${_include_dir}/petsc.h")
-        set (PETSC_INCLUDE_DIR "${_include_dir}" CACHE INTERNAL "Internal PETSc include directory" FORCE)
-        break ()
-      endif ()
-    endforeach ()
+# libm is always required
+list(APPEND _petsc_libs "${SUNDIALS_MATH_LIBRARY}")
 
-    # check if the include directory was found
-    if (NOT PETSC_INCLUDE_DIR)
-      string (CONCAT msg
-        "Could not determine PETSc include directory from PETSC_INCLUDES:\n"
-        "  PETSC_INCLUDES=${PETSC_INCLUDES}\n")
-      message (FATAL_ERROR ${msg})
+# Substitute MPI target if PETSC is built with MPI
+foreach(_next_lib IN LISTS PKG_PETSC_STATIC_LIBRARIES)
```

Comment:
Are these libraries always in `PKG_PETSC_STATIC_LIBRARIES`? Could they also be in `PKG_PETSC_LIBRARIES`?

Reply:
No they wont be in `PKG_PETSC_LIBRARIES`, that only will contain PETSc libraries, no dependencies.

Reply:
They will be in `PKG_PETSC_STATIC_LIBRARIES` if PETSc was built with MPI and Kokkos respectively.